packages/pyright-internal/src/languageService/hoverProvider.ts

```diff
@@ -648,98 +590,97 @@ export class HoverProvider {
                     parts.push({ text: '---\n' });
                 }
 
                 this._addResultsPart(parts, markDown);
-            } else if (format === MarkupKind.PlainText) {
+            } else if (this._format === MarkupKind.PlainText) {
                 this._addResultsPart(parts, convertDocStringToPlainText(docString));
             } else {
-                fail(`Unsupported markup type: ${format}`);
+                fail(`Unsupported markup type: ${this._format}`);
             }
         }
     }
 
-    private static _addResultsPart(parts: HoverTextPart[], text: string, python = false) {
+    private _addResultsPart(parts: HoverTextPart[], text: string, python = false) {
         parts.push({
             python,
             text,
         });
     }
-}
 
-export function convertHoverResults(
-    format: MarkupKind,
-    hoverResults: HoverResults | undefined,
-    includeHash?: boolean
-): Hover | undefined {
-    if (!hoverResults) {
-        return undefined;
-    }
+    private _convertHoverResults(hoverResults: HoverResults | undefined): Hover | undefined {
+        if (!hoverResults) {
+            return undefined;
+        }
 
-    let markupString = hoverResults.parts
-        .map((part) => {
-            if (part.python) {
-                if (format === MarkupKind.Markdown) {
-                    return '```python\n' + part.text + '\n```\n';
-                } else if (format === MarkupKind.PlainText) {
-                    return part.text + '\n\n';
-                } else {
-                    fail(`Unsupported markup type: ${format}`);
+        let markupString = hoverResults.parts
+            .map((part) => {
+                if (part.python) {
+                    if (this._format === MarkupKind.Markdown) {
+                        return '```python\n' + part.text + '\n```\n';
+                    } else if (this._format === MarkupKind.PlainText) {
+                        return part.text + '\n\n';
+                    } else {
+                        fail(`Unsupported markup type: ${this._format}`);
+                    }
                 }
-            }
-            return part.text;
-        })
-        .join('')
-        .trimEnd();
-
-    // If we have a lastKnownModule in the hover results, stick in a comment with
-    // the hashed module name. This is used by the other side to send telemetry.
-    if (hoverResults.lastKnownModule && format === MarkupKind.Markdown && includeHash) {
-        markupString += `\n<!--moduleHash:${hashString(hoverResults.lastKnownModule)}-->`;
+                return part.text;
+            })
+            .join('')
+            .trimEnd();
+
+        // If we have a lastKnownModule in the hover results, stick in a comment with
+        // the hashed module name. This is used by the other side to send telemetry.
+        if (hoverResults.lastKnownModule && this._format === MarkupKind.Markdown && this._supportTelemetry) {
```

Comment:
since `hoverprovider` is instance class now, we can do this only in `pylance` and not leak this into `pyright`

Reply:
You mean we can subclass it, right? Or are we thinking of just forking the whole thing?

Reply:
subclass.